pallets/rmrk-core/src/lib.rs

```diff
@@ -354,18 +356,45 @@ pub mod pallet {
 		) -> DispatchResult {
 			let sender = ensure_signed(origin.clone())?;
 
+			// Get current owner for child removal later
+			let parent =
+				pallet_uniques::Pallet::<T>::owner(collection_id, nft_id);
+			// Check if parent returns None which indicates the NFT is not available
+			ensure!(parent.is_some(),Error::<T>::NoAvailableNftId);
+			let current_owner = parent.as_ref().unwrap();
+
 			let max_recursions = T::MaxRecursions::get();
-			Self::nft_send(
+			let new_owner_account = Self::nft_send(
 				sender.clone(),
 				collection_id,
 				nft_id,
 				new_owner.clone(),
 				max_recursions,
 			)?;
 
+			pallet_uniques::Pallet::<T>::do_transfer(
+				collection_id,
+				nft_id,
+				new_owner_account.clone(),
+				|class_details, details|
+					Ok(())
+			)?;
+
+			// Handle Children StorageMap for NFTs
+			let current_cid_nid =  Pallet::<T>::decode_nft_account_id::<T::AccountId>(current_owner.clone());
+			if current_cid_nid.is_some() {
+				// remove child from parent
```

Comment:
What if the new owner is not a NFT? Should we still keep tracking the children? It may be on very large for a big player

Reply:
Would we need to change the `Children` StorageMap or would this be tracked in `NftsByOwner` StorageMap? Right now children are tracked for NFTs that own NFTs.

Reply:
Being tracked in Issue #48